lib/perl/Genome/Model/Tools/DetectVariants2/Dispatcher.pm

```diff
@@ -932,40 +938,39 @@ sub _create_temp_directories {
 # After promoting staged data as per normal, create a symlink from the final output files to the top level of the dispatcher output directory
 sub _promote_staged_data {
     my $self = shift;
-    my $output_dir  = $self->output_directory;
 
     # This sifts the workflow results for relative paths to output files, and places them in the appropriate params
     $self->set_output_files($self->_workflow_result);
 
     # Symlink the most recent version bed files of the final hq calls into the base of the output directory
-    for my $variant_type (@{$self->variant_types}){
-        my $output_accessor = "_".$variant_type."_hq_output_file";
-        if(defined($self->$output_accessor)){
+    for my $variant_type ( @{$self->variant_types} ) {
+        my $output_accessor = '_'. $variant_type .'_hq_output_file';
+        if ( defined($self->$output_accessor) ) {
             my $file = $self->$output_accessor;
             my $output_file = basename($file);
-            my $output = "$output_dir/$output_file";
+            my $output = File::Spec->join($self->_temp_staging_directory,$output_file);
             if (-e $file) {
                 Genome::Sys->create_symlink($file,$output);
             }
 
             # This may or may not exist, depending on the variant type
-            my $vcf_link = dirname($file)."/$variant_type" . "s.vcf.gz";
-            if(-e $vcf_link){
+            my $vcf_link = File::Spec->join(dirname($file),$variant_type .'s.vcf.gz');
+            if (-e $vcf_link){
                 my $source;
-                if(-l $vcf_link){
+                if (-l $vcf_link) {
                     $source = readlink($vcf_link);
                 } else {
                     $source = $vcf_link;
                 }
-                my $link_target = $output_dir."/$variant_type" . "s.detailed.vcf.gz";
-                my $clipped_vcf = $output_dir."/$variant_type" . "s.vcf.gz";
+                my $link_target = File::Spec->join($self->_temp_staging_directory,$variant_type .'s.detailed.vcf.gz');
+                my $clipped_vcf = File::Spec->join($self->_temp_staging_directory, $variant_type .'s.vcf.gz');
```

Comment:
I don't understand the logic behind making this change.  The final product is a symlink to a location in the `output directory`; it seems merely coincidental if this happens to work when created as a symlink to the `_temp_staging_directory` instead.